pkg/strace/internal/binary/binary_test.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2024 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package binary
+
+import (
+	"encoding/binary"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+type ts struct {
+	I1 int8
+	I2 int16
+	I3 int32
+	I4 int64
+	S  tts
+	A  [1]int32
+}
+
+type tts struct {
+	U1 uint8
+	U2 uint16
+	U3 uint32
+	U4 uint64
+	A  [1]int32
+}
+
+func TestMarshalUnmarshal(t *testing.T) {
```

Comment:
would it make sense to use a table based testing to list out more cases, and including littleEndian ? 

e.g. 

```
for _, tt := range struct {
    name string  // test case name
    t1       ts  // original data
    order  binary.ByteOrder 
} {
    {
       name: "little endian, basic types, no nested struct, empty array",
       t1: ts{
		I1: 1,
		I2: -2,
		I3: 3,
		I4: -4,
                S: nil,
                A: nil,
        },
       order: binary.LittleEndian,
    },
    {
      name: "little endian, basic types, nested struct, empty array",
      t1: ts{
		I1: 1,
		I2: -2,
		I3: 3,
		I4: -4,
		S: tts{
			U1: 10,
			U2: 11,
			U3: 12,
			U4: 13,
		},
                A: nil,
           },
    },
    ... // more cases go here
} {
     t.Run(tt.name, func(t testing.T){
         mb := Marshal(nil, tt.order, tt.t1)
         var t2 ts
         Unmarshal(mb, binary.BigEndian, &t2)
         diff := cmp.Diff(t1, t2)
	if diff != "" {
		t.Errorf(" t1 is not equal to t2:\n%s", diff)
	}
     })
}

```

Reply:
Yeah good idea, I added endianness and more extreme values for some fields. For table driven tests we probably want different structures as payload for marshal/unmarshal, but with any it became more complicated to unmarshal bytes back, so I think it better to leave it as is.

Reply:
Not quite follow how it can become more complicated. Elaborate ?

ack this change is net positive

Reply:
For table driven test I see more value if I could change ts structure from test to test, like only basic types, basic types + structure, etc. In this case I would need to change ts to any, but then I can't simply write `var t2 any` and unmarshal bytes back. I think it possible with reflection but a bit awkward. Also we can't write `S: nil` it will not compile.

That's why I initially decide to create big enough structure to cover all cases.

Reply:
I don't understand how a byte slice helps. I need a type to check if type1 equals unmarshall(marshall(type1)), and to do that, it is quite handy to have a type.

Reply:
resolving for now, if needed we can add more test later.